### PR TITLE
[SPARK-6451][SQL] supported code generation for CombineSum

### PR DESCRIPTION
**Pull request by gvramana**

### CODE REVIEW

**Comment by gvramana** [comment]
@marmbrus , @yhuai Please review the same. Thanks

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala [hunk 1]
```diff
@@ -220,6 +220,19 @@ class SQLQuerySuite extends QueryTest with BeforeAndAfterAll {
...
+    checkAnswer(
+      sql("SELECT a, SUM(b), SUM(CAST(b as Double)) FROM testData2 GROUP BY a"),
+      Seq(Row(1, 3, 3.0), Row(2, 3, 3.0), Row(3, 3, 3.0)))
+
+    checkAnswer(
+      sql("SELECT  sum('a'), avg('a') FROM testData"),
+      Row(0, null))
+    setConf(SQLConf.CODEGEN_ENABLED, originalValue.toString)      
+  }
+
```
Comment (yhuai) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala]: I also have some tests added in https://github.com/yhuai/spark/blob/expressionType/sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala#L107. Can you combine my stuff into your test?  I think we can use the test of "aggregation with codegen" for now. Also, I think it is necessary to check the plan to make sure we indeed have GeneratedAggregate.

**Comment by gvramana** [comment]
ok, I will combine them.

**Comment by gvramana** [comment]
@yhuai Combined code generation tests and updated sum aggregation test accordingly. Thanks

**Comment by marmbrus** [comment]
ok to test

**Comment by gvramana** [comment]
Test Failure does not look related to fix. Please check.

**Comment by gvramana** [comment]
rebased code to latest. please test the same.

**Comment by marmbrus** [comment]
Thanks, merged to master!

**Pull request closed without merging**